src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_method.yml

```diff
@@ -26,10 +26,12 @@ sylius_grid:
                     label: sylius.ui.name
                     sortable: translation.name
                 gateway:
-                    type: string
-                    path: gatewayConfig.gatewayName
+                    type: twig
+                    path: gatewayConfig.factoryName
```

Comment:
I am not sure but it worth to menition that the gateway's factoryName property and its getter and setter are [deprecated](https://github.com/Payum/Payum/blob/bc09fd5f2c4561b6c42dc913bbdebb98134cad6e/src/Payum/Core/Model/GatewayConfigInterface.php#L21)

The factory name is inside the gateway's config. the key is `factory` (similar as in the payum.yml)